great_expectations/core/batch_manager.py

```diff
@@ -0,0 +1,169 @@
+        """Dictionary of loaded BatchData objects."""
+        return self._batch_data_cache
+
+    @property
+    def loaded_batch_ids(self) -> List[str]:
+        """IDs of loaded BatchData objects."""
+        return list(self._batch_data_cache.keys())
+
+    @property
+    def active_batch_data_id(self) -> Optional[str]:
+        """The batch id for the default batch data.
+
+        When a specific Batch objec is unavailable, then the data associated with the active_batch_data_id will be used.
+        """
+        if self._active_batch_data_id is not None:
+            return self._active_batch_data_id
+
+        if len(self._batch_data_cache) == 1:
+            return list(self._batch_data_cache.keys())[0]
+
+        return None
+
+    @property
+    def active_batch_data(self) -> Optional[BatchData]:
+        """The BatchData object from the currently-active Batch object."""
+        if self.active_batch_data_id is None:
+            return None
+
+        return self._batch_data_cache.get(self.active_batch_data_id)
+
+    @property
+    def batch_cache(self) -> Dict[str, Batch]:
+        """Getter for ordered dictionary (cache) of "Batch" objects in use (with batch_id as key)."""
+        if not isinstance(self._batch_cache, OrderedDict):
+            self._batch_cache = OrderedDict(self._batch_cache)
+
+        return self._batch_cache
+
+    @property
+    def active_batch_id(self) -> Optional[str]:
+        """Getter for active Batch ID"""
+        active_batch_data_id: str = self._active_batch_data_id
+        if active_batch_data_id != self._active_batch_id:
+            logger.warning(
+                "ID of active Batch and ID of active loaded BatchData differ."
+            )
+
+        return self._active_batch_id
+
+    @property
+    def active_batch(self) -> Optional[Batch]:
+        """Getter for active Batch"""
+        active_batch_id: Optional[str] = self.active_batch_id
+        batch: Optional[Batch] = (
+            None if active_batch_id is None else self.batch_cache.get(active_batch_id)
+        )
+        return batch
+
+    @property
+    def active_batch_spec(self) -> Optional[BatchSpec]:
+        """Getter for active batch's batch_spec"""
+        if not self.active_batch:
+            return None
+
+        return self.active_batch.batch_spec
+
+    @property
+    def active_batch_markers(self) -> Optional[BatchMarkers]:
+        """Getter for active batch's batch markers"""
+        if not self.active_batch:
+            return None
+        else:
+            return self.active_batch.batch_markers
+
+    @property
+    def active_batch_definition(self) -> Optional[BatchDefinition]:
+        """Getter for the active batch's batch definition"""
+        if not self.active_batch:
+            return None
+        else:
+            return self.active_batch.batch_definition
+
+    def reset_batch_cache(self) -> None:
+        """Clears Batch cache"""
+        self._batch_cache = {}
```

Comment:
Can we make this an ordered dict since it is converted upon retrieval?

Reply:
@billdirks Will try in the next commit -- thanks!